__tests__/main.test.js

```diff
@@ -1,8 +1,73 @@
-const { _ } = require('higher-order-.js');
+const { _ } = require('../higher-order.js');
 
 describe('_.map', () => {
-    test('it maps every item in the provided array and returns the mapped output', () => {
+    test('throws a typeError if the provided object to be mapped isn\'t an array', () => {
+        let input = null;
+        let inputMapper = null;
 
+        expect(() => {
+            _.map(input, inputMapper);
+        }).toThrow();
     });
-});
 
+    test('throws a typeError if the provided mapper isn\'t a function value', () => {
+        let input = [];
+        let inputMapper = null;
+
+        expect(() => {
+            _.map(input, inputMapper);
+        }).toThrow();
+    });
+
+    test('maps an empty array and returns an empty array object', () => {
+        let input = [];
+        let inputMapper = () => {};
+        let expectedOutput = [];
+
+        expect(_.map(input, inputMapper)).toEqual(expectedOutput);
+    });
+
+    test('maps every item in the provided array and returns the mapped output', () => {
+        let input = [0, 1, 2, 3, 4, 5];
+        let inputMapper = (value) => {
```

Comment:
```javascript
const inputMapper = (length) => '#'.repeat(length);
```

Reply:
Or point-free style (wouldn't recommend this, just showing it because we're thinking about higher-order functions):

```javascript
const inputMapper = String.prototype.repeat.bind('#')
```

Reply:
Much more readable doing this with the `repeat` function.